Mathlib/Algebra/Ring/Divisibility/Basic.lean

```diff
@@ -36,6 +36,23 @@ theorem MulEquiv.decompositionMonoid (f : F) [DecompositionMonoid β] : Decompos
     iterate 2 erw [(f : α ≃* β).apply_symm_apply]
     exact h
 
+/--
+If `G` is a `LeftCancelSemiGroup`, left multiplication by `g` yields an equivalence between `G`
+and the set of elements of `G` divisible by `g`.
+-/
+noncomputable def Equiv.dvd {G : Type*} [LeftCancelSemigroup G] (g : G) :
```

Comment:
I would make it `protected`. BTW, is it a special case of `Equiv.ofInjective`?

Reply:
Yes, I thought about using `Equiv.ofInjective` but it is not imported in this file and I didn't want to change the imports of such a low-level file (furthermore,  it does not make the code that much simpler).

I have `protected` the name of the function.